mmgen/core/hooks/ema_hook.py

```diff
@@ -61,14 +66,49 @@ def __init__(self,
         assert hasattr(
             self, interp_mode
         ), f'Currently, we do not support {self.interp_mode} for EMA.'
-        self.interp_func = partial(
-            getattr(self, interp_mode), **self.interp_cfg)
+        self.interp_func = getattr(self, interp_mode)
+
+        self.momentum_cfg = dict() if momentum_cfg is None else deepcopy(
+            momentum_cfg)
+        self.momentum_policy = momentum_policy
+        if momentum_policy != 'fixed':
+            assert hasattr(
```

Comment:
I suggest that we can add a class attribute called `_supported_momentum_policy`. Use instance attributes' names to check the policy may not be safe.

Reply:
True, may be we can register `interp_func` and `momentum_updater` in this class?